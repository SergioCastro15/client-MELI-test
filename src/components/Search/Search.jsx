import React, { useState, useContext } from 'react'
import searchContext from '../../context/context'
import { useHistory } from 'react-router-dom'
import logo from '../../logo.png'
import lupa from '../../lupa.png'

import './Search.scss';

export const Search = () => {
    const history = useHistory();
    const { getCurrentSearch } = useContext(searchContext)

    const [textValue, setTextValue] = useState({
        search: ''
    })

    const triggerInputSearch = (e) => {
        e.preventDefault()
        setTextValue({
            ...textValue,
            [e.target.name]: e.target.value
        })
    }

    const activeSearchButton = (e) => {
        e.preventDefault()
        getCurrentSearch(textValue.search)
        history.push('/items?q=' + textValue.search)
    }

    return (
        <nav className="navbar-search">
            <img className="logo" src={logo} alt="logo"/>
            <form onSubmit={activeSearchButton} >
                <input
                    className="input-search"
                    name="search"
                    onChange={triggerInputSearch} 
                    placeholder='Nunca dejes de buscar' 
                    type="text"
                />
                <button type="submit">
                    <img className="lupa" src={lupa} alt="lupa"/>
                </button>
            </form>

        </nav>
    )
}
