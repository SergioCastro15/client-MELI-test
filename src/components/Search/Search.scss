@import '../../scss/variables';

.navbar-search {
    background: $yellow;
    height: 45px;
    display: flex;
    justify-content: center;
    align-items: center;

    form{
        .input-search {
            width: 700px;
            height: 30px;
            border: none;
        }

        button {
            border: none;
            height: 33px;
            width: 50px;
    
            .lupa {
                background: $gray-light;
                height: 15px;
            }
    
            &:hover {
                cursor: pointer;
            }
        }
    }


    .logo{
        height: 30px;
        margin-right: 15px;
    }
}

::placeholder {
    color: $gray;
    opacity: 1;
}

input{
    text-indent: 10px;
}

// breakpoints 

@media (max-width: 768px) {
    .navbar-search {
        form{
            .input-search {
                width: 250px;
                height: 32px;
            }
    
            button {
                height: 35px;
            }
        }

        .logo{
            height: 25px;
        }
    }
}