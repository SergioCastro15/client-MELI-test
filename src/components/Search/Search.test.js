import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Search } from './Search'
import SearchState from '../../context/state'
import SearchContext from '../../context/context'

import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'

const renderWithContext = (component) => {
    const history = createMemoryHistory()

    return {
        ...render(
            <SearchState value={SearchContext}>
                <Router history={history}>
                    {component}
                </Router>
            </SearchState>
        ),history
    }
}

it('should take a snapshot', () => {
    const { asFragment } = renderWithContext(<Search />)
    
    expect(asFragment(<Search />)).toMatchSnapshot()
})


it('should navigate to home page', ()=> {
    const { getByTestId, history } = renderWithContext(<Search />) 
    fireEvent.click(getByTestId('go-home'))

    expect(history.location.pathname).toBe("/")
})

it('should navigate to items page', () => {
    const { getByTestId, history } = renderWithContext(<Search />) 
    fireEvent.submit(getByTestId('go-items'))

    expect(history.location.pathname).toBe("/items")
})