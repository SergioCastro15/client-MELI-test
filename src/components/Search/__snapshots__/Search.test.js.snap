// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar-search"
  >
    <img
      alt="logo"
      class="logo"
      data-testid="go-home"
      src="logo.png"
    />
    <form
      data-testid="go-items"
    >
      <input
        class="input-search"
        name="search"
        placeholder="Nunca dejes de buscar"
        type="text"
      />
      <button
        type="submit"
      >
        <img
          alt="lupa"
          class="lupa"
          src="lupa.png"
        />
      </button>
    </form>
  </nav>
</DocumentFragment>
`;
